io/zenoh-transport/src/unicast/manager.rs

```diff
@@ -406,12 +422,53 @@ impl TransportManager {
                     #[cfg(feature = "shared-memory")]
                     is_shm: config.is_shm,
                 };
-                let a_t = Arc::new(TransportUnicastInner::make(self.clone(), stc)?);
+
+                // select and create transport implementation depending on the cfg
+                let a_t: Result<Arc<dyn TransportUnicastInnerTrait>, (Error, Option<u8>)> = {
+                    #[cfg(feature = "shared-memory")]
+                    {
+                        match stc.is_shm {
+                            false => {
+                                log::info!("Will use NET transport!");
+                                let t: Arc<dyn TransportUnicastInnerTrait> =
+                                    TransportUnicastInner::make(self.clone(), stc)
+                                        .map_err(|e| (e, Some(close::reason::INVALID)))
+                                        .map(|v| {
+                                            Arc::new(v) as Arc<dyn TransportUnicastInnerTrait>
+                                        })?;
+                                // Add the link to the transport
+                                t.add_link(link, direction)
+                                    .map_err(|e| (e, Some(close::reason::MAX_LINKS)))?;
+                                Ok(t)
+                            }
+                            true => {
+                                log::info!("Will use SHM transport!");
+                                Ok(ShmTransportUnicastInner::make(self.clone(), stc, link)
+                                    .map_err(|e| (e, Some(close::reason::INVALID)))
+                                    .map(|v| Arc::new(v) as Arc<dyn TransportUnicastInnerTrait>)?)
+                            }
+                        }
+                    }
+                    #[cfg(not(feature = "shared-memory"))]
+                    {
+                        log::info!("Will use the only NET transport!");
+                        let t: Arc<dyn TransportUnicastInnerTrait> =
```

Comment:
There is some code duplication introduced by the conditional compilation. Need to investigate a patter on how to remove that.